File: zzyl-service/src/main/java/com/zzyl/entity/Leave.java
package com.zzyl.entity;

import com.zzyl.base.BaseEntity;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import org.apache.ibatis.type.Alias;

import java.time.LocalDateTime;

/**
 * 请假实体类
 */
@Data
@ApiModel(description = "请假实体类")
public class Leave extends BaseEntity {

    /**
     * 请假单号
     */
    @ApiModelProperty(value = "请假单号")
    private String leaveCode;

    /**
     * 请假标题
     */
    @ApiModelProperty(value = "请假标题")
    private String title;

    /**
     * 老人id
     */
    @ApiModelProperty(value = "老人id")
    private Long elderId;

    /**
     * 老人姓名
     */
    @ApiModelProperty(value = "老人姓名")
    private String name;

    /**
     * 身份证号
     */
    @ApiModelProperty(value = "身份证号")
    private String idCardNo;

    /**
     * 联系方式
     */
    @ApiModelProperty(value = "联系方式")
    private String phone;

    /**
     * 护理等级
     */
    @ApiModelProperty(value = "护理等级")
    private String nursingLevelName;

    /**
     * 床位编号
     */
    @ApiModelProperty(value = "床位编号")
    private String bedNo;

    /**
     * 养老顾问
     */
    @ApiModelProperty(value = "养老顾问")
    private String counselor;

    /**
     * 请假开始时间
     */
    @ApiModelProperty(value = "请假开始时间")
    private LocalDateTime leaveStartTime;

    /**
     * 请假结束时间
     */
    @ApiModelProperty(value = "请假结束时间")
    private LocalDateTime leaveEndTime;

    /**
     * 请假原因
     */
    @ApiModelProperty(value = "请假原因")
    private String leaveReason;

    /**
     * 备注
     */
    @ApiModelProperty(value = "备注")
    private String remark;

    /**
     * 申请人
     */
    @ApiModelProperty(value = "申请人")
    private String applicat;

    /**
     * 申请人部门编号
     */
    @ApiModelProperty(value = "申请人部门编号")
    private String deptNo;

    /**
     * 申请人id
     */
    @ApiModelProperty(value = "申请人id")
    private Long applicatId;

    /**
     * 申请时间
     */
    @ApiModelProperty(value = "申请时间")
    private LocalDateTime createTime;

    /**
     * 流程状态
     * 0:申请请假
     * 1:护理组长审批
     */
    @ApiModelProperty(value = "流程状态")
    private Integer flowStatus;

    /**
     * 状态（1：申请中，2:已完成,3:已关闭）
     */
    @ApiModelProperty(value = "状态")
    private Integer status;

    // 流程状态枚举
    @Alias("LeaveFlowStatus")
    public enum FlowStatus {
        APPLY(0, "申请请假"),
        NURSE_APPROVAL(1, "护理组长审批");

        Integer code;
        String name;

        FlowStatus(Integer code, String name) {
            this.code = code;
            this.name = name;
        }

        public Integer getCode() {
            return this.code;
        }

        public String getName() {
            return this.name;
        }
    }

    // 状态枚举
    @Alias("LeaveStatus")
    public enum Status {
        APPLICATION(1),
        FINISHED(2),
        CLOSED(3);

        Integer code;

        Status(Integer code) {
            this.code = code;
        }

        public Integer getCode() {
            return this.code;
        }
    }
}

File: zzyl-ui/project-zhyl-admin-vue3-java/src/api/leave.ts
import { request } from '@/utils/request'
import type { ListResult, ListModel, FormApply } from '@/api/model/synergyModel'

// ========== 类型定义 ==========

// 请假列表项
export interface LeaveListModel {
  id: number
  leaveCode: string
  title: string
  elderId: number
  name: string
  idCardNo: string
  phone: string
  nursingLevelName: string
  bedNo: string
  counselor: string
  leaveStartTime: string
  leaveEndTime: string
  leaveReason: string
  remark: string
  applicat: string
  deptNo: string
  applicatId: number
  createTime: string
  flowStatus: number
  status: number
  taskId: string
}

// 请假表单提交
export interface FormLeave {
  leaveCode: string
  title: string
  elderId: number
  name: string
  idCardNo: string
  phone: string
  nursingLevelName: string
  bedNo: string
  counselor: string
  leaveStartTime: string
  leaveEndTime: string
  leaveReason: string
  remark: string
  applicat: string
  deptNo: string
  applicatId: number
}

// 请假操作参数
export interface LeaveActionParams {
  leaveCode: string
  message?: string
  reject?: string
  flowStatus?: number
  taskId: string
}

// 请假分页查询参数
export interface LeaveQueryParams {
  leaveCode?: string
  name?: string
  idCardNo?: string
  startTime?: number
  endTime?: number
  pageNum?: number
  pageSize?: number
}

// 请假响应
export interface LeaveResult {
  data: any[]
  list: Array<LeaveListModel>
}

// ========== 请假管理 API ==========

// 申请请假
export function createLeave(params: FormLeave) {
  return request.post<ListResult>({
    url: '/leave/create',
    data: params
  })
}

// 请假表单查询
export function getLeave(params: {
  leaveCode: string
  assigneeId?: string
  flowStatus: number
  taskId?: string
}) {
  return request.get<FormApply>({
    url: '/leave',
    params
  })
}

// 提交请假
export function submitLeave(params: { code: string; assigneeId?: string; taskId?: string }) {
  return request.post<ListResult>({
    url: '/leave/submit',
    data: params
  })
}

// 驳回
export function disapproveLeave(params: LeaveActionParams) {
  return request.put<ListResult>({
    url: `/leave?leaveCode=${params.leaveCode}&message=${params.message}&taskId=${params.taskId}`
  })
}

// 审核拒绝
export function rejectLeave(params: LeaveActionParams) {
  return request.put<ListResult>({
    url: `/leave/reject?leaveCode=${params.leaveCode}&reject=${params.reject}&taskId=${params.taskId}`
  })
}

// 撤回
export function revocationLeave(params: LeaveActionParams) {
  return request.put<ListResult>({
    url: `/leave/revocation?leaveCode=${params.leaveCode}&flowStatus=${params.flowStatus}&taskId=${params.taskId}`
  })
}

// 撤销
export function cancelLeave(params: LeaveActionParams) {
  return request.put<ListResult>({
    url: `/leave/cancel?leaveCode=${params.leaveCode}&taskId=${params.taskId}`
  })
}

// 请假分页查询
export function getLeavePage(params: LeaveQueryParams) {
  return request.post<LeaveResult>({
    url: '/leave/selectByPage',
    data: params
  })
}
